pkg/validator/am0001/default_channel.go

```diff
@@ -84,3 +90,38 @@ func (d *DefaultChannel) isListedInChannels(channels *[]v1alpha1.Channel, defaul
 	msg := fmt.Sprintf("The defaultChannel '%v' is not part of the listed channelNames: %v.", defaultChannel, channelNames)
 	return d.Fail(msg)
 }
+
+func (d *DefaultChannel) isListedInBundleChannel(defaultChannel string, bundles []*registry.Bundle) validator.Result {
+	var message []string
+	for _, bundle := range bundles {
+		if bundle.Annotations.DefaultChannelName != defaultChannel && bundle.Annotations.DefaultChannelName != "" {
+			msg := fmt.Sprintf("The defaultChannel is '%v' is not matching operators.operatorframework.io.bundle.channel.default.v1 '%v'.",
+				defaultChannel, bundle.Annotations.DefaultChannelName,
+			)
+			message = append(message, msg)
+		}
+
+		channels := strings.Split(bundle.Annotations.Channels, ",")
+
+		isPresentInBundleChannels := func(defaultChannel string, channels []string) bool {
```

Comment:
Here is a good article about inner functions https://realpython.com/inner-functions-what-are-they-good-for/#using-inner-vs-private-helper-functions. It is about python but I think the same applies to golang. It also says when it is just a helper function, it is probably better to define it outside the other function.

Reply:
Also, the function is only called once, so I don't necessarily think the code has to be pulled out into a different function :)

Reply:
Actually, It was kind of easy to do it there only. That's why I didn't created another function outside.

Reply:
I don't like inner functions used like this since they provide a net-neutral benefit to readability (code block is contained and semantically named, but outer function is still too big). I think the more idiomatic way to do this in Go would be to move the functionality into a `type`.

Something like the following:

```go
func channelListFromAnnotiation(ann string) channelList {
	return channelList(strings.Split(",", ann))
}

type channelList []string

func (l channelList) Contains(ch string) bool {
	for _, c := range l {
		if ch == c {
			return true
		}
	}

	return false
}

func (l channelList) String() string {
	return strings.Join(l, ", ")
}
```

Then you have have stronger cohesion in the domain of `Channels`.

Reply:
Okay. I will do it.

Reply:
Defining a function, especially a inner function, does add some complexity. I think it might be cleaner just to write

`
defaultChannelPresent := false
for _, channel := range channels {
if defaultChannel == channel {
	defaultChannelPresent := false
        break
}
if !defaultChannelPresent {
`

Reply:
WDYT?

Reply:
I was also thinking the same in the first place but, used inner functions instead for the same.